### PR TITLE
Bump upjet to the latest commit

### PR DESCRIPTION
**Pull request by sergenyalcin**
### Description of your changes

This PR bumps the upjet dependency to the latest commit

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

examples/applications/v1beta2/application.yaml: https://github.com/crossplane-contrib/provider-upjet-azuread/actions/runs/9111424565

examples/conditionalaccess/v1beta2/accesspolicy.yaml: https://github.com/crossplane-contrib/provider-upjet-azuread/actions/runs/9111126161

examples/conditionalaccess/v1beta2/location.yaml: https://github.com/crossplane-contrib/provider-upjet-azuread/actions/runs/9111541469

examples/groups/v1beta2/group.yaml: https://github.com/crossplane-contrib/provider-upjet-azuread/actions/runs/9111610387

examples/serviceprincipals/v1beta2/principal.yaml:

examples/groups/v1beta1/member.yaml:

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by sergenyalcin** [comment]
/test-examples="examples/applications/v1beta2/application.yaml"

**Comment by sergenyalcin** [comment]
/test-examples="examples/applications/v1beta2/application.yaml"

**Comment by sergenyalcin** [comment]
/test-examples="examples/applications/v1beta2/application.yaml"

**Comment by sergenyalcin** [comment]
/test-examples="examples/conditionalaccess/v1beta2/accesspolicy.yaml"

**Comment by sergenyalcin** [comment]
/test-examples="examples/applications/v1beta2/application.yaml"

**Comment by sergenyalcin** [comment]
/test-examples="examples/conditionalaccess/v1beta2/location.yaml"

**Comment by sergenyalcin** [comment]
/test-examples="examples/groups/v1beta2/group.yaml"

**Comment by sergenyalcin** [comment]
/test-examples="examples/serviceprincipals/v1beta2/principal.yaml"

**Comment by sergenyalcin** [comment]
/test-examples="examples/groups/v1beta1/member.yaml"

File: config/provider.go [hunk 1]
```diff
@@ -139,10 +139,11 @@ func bumpVersionsWithEmbeddedLists(pc *ujconfig.Provider) {
 		// we would like to set the storage version to v1beta1 to facilitate
 		// downgrades.
 		r.SetCRDStorageVersion("v1beta1")
+		r.ControllerReconcileVersion = "v1beta1"
 		r.Conversions = []conversion.Conversion{
 			conversion.NewIdentityConversionExpandPaths(conversion.AllVersions, conversion.AllVersions, []string{"spec.forProvider", "spec.initProvider", "status.atProvider"}, r.CRDListConversionPaths()...),
```
Comment (ulucinar) [inline config/provider.go:144]: nit: We can also update the string literals for the three API roots here. But let's keep the tested state and we can do this in a follow-up PR if you also prefer.

**Review by ulucinar — APPROVED** [review]
Thanks @sergenyalcin, lgtm.

**Pull request merged**